### ISSUE TITLE
Error on carrot service stopping

### ISSUE THREAD

**Issue by mihasK** [chris104957/django-carrot#104]
I have the following error on Macos:
```
AttributeError: 'ChannelClosed' object has no attribute 'endswith'
(2 additional frame(s) were not displayed)
...
  File "django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "django/core/management/base.py", line 316, in run_from_argv
    self.execute(*args, **cmd_options)
  File "django/core/management/base.py", line 353, in execute
    output = self.handle(*args, **options)
  File "carrot/management/commands/carrot.py", line 211, in handle
    self.stderr.write(self.style.ERROR(err))
  File "django/core/management/base.py", line 142, in write
    if ending and not msg.endswith(ending):

ChannelClosed: 
(8 additional frame(s) were not displayed)
...
  File "carrot/consumer.py", line 532, in stop_consuming
    t.stop()
  File "carrot/consumer.py", line 405, in stop
    self.stop_consuming()
  File "carrot/consumer.py", line 380, in stop_consuming
    self.channel.basic_cancel(self.on_cancel, self._consumer_tag)
  File "pika/channel.py", line 222, in basic_cancel
    self._validate_channel_and_callback(callback)
  File "pika/channel.py", line 1362, in _validate_channel_and_callback
    raise exceptions.ChannelClosed()
```

Moreover, carrot service can't stop while running the task. Only `kill -9` helps. Is it a feature or a bug?

**Comment by chris104957** [comment]
So the first half of this exception is due to Django not being able to apply colours to the stderr output for an exception raised by pika. This has now been fixed in 1.4.0a6. However, I'd need more information to help with the second part - the channel was closed, but I'd need further information to establish why

**Comment by chris104957** [comment]
Closing due to lack of activity
